objectherkenning_openbare_ruimte/convert_dataset_pipeline/source/equirectangular_to_cubemap_converter.py

```diff
@@ -0,0 +1,752 @@
+        br_star = (br_x, br_y)
+
+        return tl_star, br_star
+
+    @staticmethod
+    def _reproject_xy_coor_to_face(
+        coor_xy_face: np.ndarray, coor: np.ndarray, face_w: int
+    ) -> Tuple[int, int]:
+        """
+        Compute a pair of coordinates (y_proj, x_proj) that indicate the position
+        on the cube face closest to the original equirectangular point.
+
+        To do that, we first compute the difference between each point's UV coordinates
+        on the cube face (coor_xy_face) and the original point's equirectangular coordinates (coor).
+        The difference is a 2D vector for each point on the cube face that indicates how far
+        and in which direction each cube face point is from the original equirectangular point.
+
+        np.linalg.norm(coor_xy_face - coor, axis=-1) calculates the Euclidean distance (norm)
+        between the original point and each point on the cube face.
+
+        np.argmin(euclidean_distance) finds the index of the minimum value in the array of distances
+        calculated in the previous step. This index corresponds to the location on the cube face
+        that is closest to the original point in equirectangular space.
+
+        np.unravel_index(min_distance_index, (face_w, face_w)) converts the flattened index back into
+        2D coordinates within the context of the cube face's dimensions.
+
+        Parameters
+        ----------
+        coor_xy_face : numpy.ndarray
+            UV coordinates of the cube face.
+        coor : numpy.ndarray
+            Equirectangular coordinates of the original point.
+        face_w : int
+            Width of the cube face.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the new coordinates within the face.
+        """
+
+        vector_distance = coor_xy_face - coor
+        euclidean_distance = np.linalg.norm(vector_distance, axis=-1)
+        min_distance_index = np.argmin(euclidean_distance)
+
+        y_proj, x_proj = np.unravel_index(min_distance_index, (face_w, face_w))
+
+        return int(y_proj), int(x_proj)
+
+    @staticmethod
+    def _adjust_coordinates_based_on_corner(
+        tag: str,
+        corner: Tuple[float, float],
+        processed_corners: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        """
+        Adjust the coordinates of a bounding box based on the processed corners within a face.
+
+        This function updates the coordinates of a bounding box in a cubemap face by considering
+        the positions of already processed corners. Depending on the corner being processed (tag),
+        it may adjust the x and y maximums or minimums of the bounding box to ensure the box
+        accurately represents the object within the constraints of the face. The function is
+        typically used in a scenario where bounding boxes may span across multiple faces of a
+        cubemap, requiring adjustments to fit within a single face properly.
```

Comment:
This explanation is also quite hard to follow. Intuitively, does this mean that a bounding box is split up into the parts that fit in separate cube faces?